### PR TITLE
ref(*): remove global mutable state in the data package

### PR DESCRIPTION
**Pull request by arschles**
doing so increases ability to test, understandability/debuggability of
the code, and concurrency safety.

See http://c2.com/cgi/wiki?GlobalVariablesAreBad for more motivation

Pre-requisite for #42

### CODE REVIEW

File: data/data.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package data
 
 import (
 	"crypto/tls"
-	"fmt"
+	// "fmt"
```
Comment (vdice) [inline data/data.go]: can probably just get rid of this comment now?

Comment (arschles) [inline data/data.go]: ah, good catch. thx

Comment (arschles) [inline data/data.go]: removed in https://github.com/deis/workflow-manager/pull/49/commits/7e50c5d6a7c6f18492dff2c019417cd3955fb63c

File: data/data.go [hunk 2]
```diff
@@ -16,55 +16,33 @@ const (
...
 
 // GetAvailableVersions gets available component version data
 func GetAvailableVersions(a AvailableVersions) ([]types.ComponentVersion, error) {
 	// First, check to see if we have an in-memory copy
-	data, err := getAvailableVersions()
+	data := a.Cached()
 	// If we don't have any cached data, get the data from the remote authority
-	if err != nil {
-		data, err = a.Refresh()
+	if len(data) == 0 {
```
Comment (vdice) [inline data/data.go]: curious as to the difference in data checks from above (here checking `len(data)`, above (line 24) checking `data == ""`)?

Comment (arschles) [inline data/data.go]: in this case, `data` is a `[]types.ComponentVersion`, and in the other case, `data` is ` string`, so the checks for a cache miss are necessarily different

Comment (vdice) [inline data/data.go]: n/m disregard -- just had to take the time to read a little more carefully :) (slice check vs. string check)

**Pull request merged**